iscript/src/iscript/macvpn.py

```diff
@@ -0,0 +1,191 @@
+
+# TODO: Move provisioning profiles from braindump
+LOGINITEM_PROVISIONING_PROFILE = "https://hg.mozilla.org/build/braindump/raw-file/tip/signing-related/firefoxvpn_loginitem_developerid.provisionprofile"
+NATIVEMESSAGING_PROVISIONING_PROFILE = "https://hg.mozilla.org/build/braindump/raw-file/tip/signing-related/FirefoxVPN_Native_Messaging.provisionprofile"
+MOZILLA_VPN_PROVISIONING_PROFILE = "https://hg.mozilla.org/build/braindump/raw-file/tip/signing-related/firefoxvpn_developerid.provisionprofile"
+
+async def _create_notarization_zipfile(work_dir, source, dest):
+    """Creates a zipfile for notarization
+
+    Keeps parent directory in zip
+
+    Args:
+        work_dir (str): base directory from the zip command perspective
+        source (str): the source directory/file
+        dest (str): destination package
+
+    Raises:
+        IScriptError: on failure
+
+    Returns:
+        str: the zip path
+    """
+    await run_command(["ditto", "-c", "-k", "--sequesterRsrc", "--keepParent", source, dest], cwd=work_dir, exception=IScriptError)
+
+
+async def _sign_app(config, task, sign_config, app, entitlements_url, provisioning_profile_url):
+    """Sign an app
+
+    Args:
+        config (dict): script config
+        task (dict): running task
+        sign_config (dict): the config for this signing key
+        app (App): the App the notarize
+        entitlements_url (str): URL for entitlements file
+        provisioning_profile_url (str): URL for provisioning profile
+
+    """
+    # We mock the task for downloading entitlements and provisioning profiles
+    entitlements_path = await download_entitlements_file(config, sign_config, {'payload': {'entitlements-url': entitlements_url}})
+    provisioning_profile_path = await download_provisioning_profile(config, {'payload': {'provisioning-profile-url': provisioning_profile_url}})
+
+    await sign_all_apps(config, sign_config, entitlements_path, [app], provisioning_profile_path)
+
+    log.info(f"Done signing app {app.app_name}")
+
+
+async def notarize_vpn_behavior(config, task):
+    """Notarization for mac vpn app.
+
+    Workflow:
+    . Sign all inner apps
+    . Sign main app
+    . Notarize main app
+
+    Args:
+        config (dict): the running configuration
+        task (dict): the running task
+
+    Raises:
+        IScriptError: on fatal error.
+
+    """
+
+    top_app = get_app_paths(config, task)
+    assert len(top_app) == 1
+
+    await extract_all_apps(config, top_app)
+
+    # Extract fills in app.parent_dir
+    top_app = top_app[0]
+
+    ##############
+    # TODO: Decide what to do with the incoming package
+    # TODO: Remove when we switch to .tar.gz payloads (1/2)
+    build_zip_path = os.path.join(top_app.parent_dir, 'BUILD.zip')
+    await run_command([
+        "unzip", build_zip_path,
+        "-d", top_app.parent_dir
+    ])
+    os.remove(build_zip_path)
+
+    top_app.app_path = os.path.join(top_app.parent_dir, "Mozilla VPN.app")
```

Comment:
Noting that we're hardcoding signing a single app and single pkg.
This matches our current requirements. If the vpn team ever needs multiple apps/pkgs signed, then we may need to take an approach like `notarize_behavior` which has a list of `App` objects and signs each of them.